gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/QuantityMapperTest.java

```diff
@@ -0,0 +1,182 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.util.ResourceUtils.getFile;
+
+import static uk.nhs.adaptors.pss.translator.util.XmlUnmarshallUtil.unmarshallFile;
+
+import org.hl7.fhir.dstu3.model.Quantity;
+import org.hl7.v3.Value;
+import org.junit.jupiter.api.Test;
+
+import lombok.SneakyThrows;
+
+public class QuantityMapperTest {
+    private static final String XML_RESOURCES_BASE = "xml/Quantity/";
+    private static final String UNIT_SYSTEM = "http://unitsofmeasure.org";
+
+    private final QuantityMapper quantityMapper = new QuantityMapper();
+
+    @SneakyThrows
+    private Value unmarshallValueElement(String fileName) {
+        return unmarshallFile(getFile("classpath:" + XML_RESOURCES_BASE + fileName), Value.class);
+    }
+
+    private void assertQuantity(
+            Quantity quantity,
+            String value,
+            String unit,
+            String system,
+            String code,
+            Quantity.QuantityComparator comparator) {
+        assertThat(quantity.getValue()).isEqualTo(value);
+        assertThat(quantity.getUnit()).isEqualTo(unit);
+        assertThat(quantity.getSystem()).isEqualTo(system);
+        assertThat(quantity.getCode()).isEqualTo(code);
+        assertThat(quantity.getComparator()).isEqualTo(comparator);
+    }
+
+    @Test
+    public void mapQuantityNoTypeStandardUnit() {
+        var value = unmarshallValueElement("no_type_standard_unit.xml");
+
+        Quantity quantity = quantityMapper.mapQuantity(value);
+
+        assertQuantity(quantity, "100", "Kg/m2", UNIT_SYSTEM, "Kg/m2", null);
+    }
+
+    @Test
+    public void mapQuantityNoTypeArbitraryUnit() {
+        var value = unmarshallValueElement("no_type_arbitrary_unit.xml");
+
+        Quantity quantity = quantityMapper.mapQuantity(value);
+
+        assertQuantity(quantity, "100", "kua/L", null, null, null);
+    }
+
+    @Test
+    public void mapQuantityPqStandardUnit() {
+        var value = unmarshallValueElement("pq_standard_unit.xml");
+
+        Quantity quantity = quantityMapper.mapQuantity(value);
+
+        assertQuantity(quantity, "100", "Kg/m2", UNIT_SYSTEM, "Kg/m2", null);
+    }
+
+    @Test
+    public void mapQuantityPqArbitraryUnit() {
+        var value = unmarshallValueElement("pq_arbitrary_unit.xml");
+
+        Quantity quantity = quantityMapper.mapQuantity(value);
+
+        assertQuantity(quantity, "100", "kua/L", null, null, null);
+    }
+
+    @Test
+    public void mapQuantityIvlPqHighStandardUnit() {
+        var value = unmarshallValueElement("ivlpq_high_standard_unit.xml");
+
+        Quantity quantity = quantityMapper.mapQuantity(value);
+
+        assertQuantity(quantity, "100", "Kg/m2", UNIT_SYSTEM, "Kg/m2", Quantity.QuantityComparator.LESS_THAN);
```

Comment:
I'm not sure if it can work, but you could try to add an import for `Quantity.QuantityComparator`, and then all of those `LESS_THAN`, `GREATER_THAN`... would be without `Quantity.QuantityComparator` prefix.

Reply:
Unfortunately the only way to have 'LESS_THAN', 'GREATER_THAN' etc with no prefix is to have an import for each of the QuantityComparator values. For now I have just imported Quantity.QuantityComparator so it looks like 'QuantityComparitor.LESS_THAN'